client/network/src/protocol/notifications/handler.rs

```diff
@@ -509,19 +563,6 @@ impl ProtocolsHandler for NotifsHandler {
 				// to do.
 				return;
 			},
-			State::Opening { ref mut in_substream, .. } |
-			State::Open { ref mut in_substream, .. } => {
-				if in_substream.is_some() {
```

Comment:
I moved these two match variants above.

If `in_substream` is `Some`, then we enter the block line 557 where we discard the new substream, because it's illegal to open a second one.

Reply:
uh, yeah - I missed that. :-)